src/libraries/Microsoft.PowerFx.Connectors/Tabular/Services/CdpDataSource.cs

```diff
@@ -32,6 +39,21 @@ public static async Task<DatasetMetadata> GetDatasetsMetadataAsync(HttpClient ht
             return await GetObject<DatasetMetadata>(httpClient, "Get datasets metadata", uri, null, cancellationToken, logger).ConfigureAwait(false);            
         }
 
+        public virtual async Task<IEnumerable<CdpTable>> GetTablesAsync(HttpClient httpClient, CancellationToken cancellationToken, ConnectorLogger logger = null)
+        {
+            if (DatasetMetadata == null)
+            {
+                DatasetMetadata = await GetDatasetsMetadataAsync(httpClient, cancellationToken, logger).ConfigureAwait(false);
+            }
+
+            string queryName = httpClient is PowerPlatformConnectorClient ppcc && ppcc.RequestUrlPrefix.Contains("/sharepointonline/") ? "/alltables" : "/tables";
```

Comment:
>httpClient is PowerPlatformConnectorClient ppcc && ppcc.RequestUrlPrefix.Contains [](http://example.com/codeflow?start=31&length=81)

The issue description in #2780 suggested dealing with this  case by looking at HttpClient.BaseAddress - would that work? It would let us avoid the type cast. 

nit-  either way - can you encapsulate this in a tiny helper method IsSharepoint(httpClient)?

Reply:
It's not in the HttpClient base address unfortunately
I will encapsulate in IsSharepoint()